docs/assignment2.md

```diff
@@ -1 +1,233 @@
+-  *winners* (type: HashMap<Integer, ArrayList<Team&gt;>):  A key represents the round number, e.g. first round is 1, while the values represent the winners at each given round.
+
+Operations
+
+- *void playRound(Queue<Team&gt; teamsToPlay, int round)*: recursive method that generates *winners*. First call is *playRound(initialTeams, 1)*. The method calls itself for following rounds until one team remains (the league winner) 
+- HashMap<Integer, ArrayList<Team>> getWinners(): calls *playRound* method and returns *winners*
+
+Associations
+
+A single **League** object can exist at one time within a **Game**. A league can start with at most 16 teams as per requirement **FR6**. Within the *playRound* method, **Match** objects are created from the playing teams to generate a winner.
+
+#### **Match**
+
+A match simulates a single soccer game between two teams.
+
+Attributes
+
+* *team1* (Type: Team): first team in the match
+* *team2* (Type: Team): second team in the match
+
+Operations
+
+- *Team getWinner()*: returns the winner of a match as described in **FR8** and **FR9**
+
+Assiciations
+
+A match is formed of exactly 2 **Team** objects.
+
+### *Views* package
+
+Each class in this package is an extention of the **GamePane** abstract class, described below. Each uses the *javafx* library and represents a different window in the GUI. All interactions with the *models* package are made via the **Game** class methods.
+
+#### **GamePane**
+
+Abstract methods
+
+- *void connectComponents()*: Connects all components into the root of a *javafx.scene.Scene*
+- *void styleComponents()*: Styles all components as required
+- *void setCallbacks()*: Sets callbacks for all interactive components
+
+### *Controllers* package
+
+Only consists of class **SceneManager** used for managing the *javafx* Scenes.
+
+Operations
+
+- *void setStage(final Stage stage)*: Sets the primary stage
+- *void showScene(final Scene scene)*: Replaces the currently active Scene with another one
+- *void showPane(final Class<? extends GamePane> pane)*: Replaces the current  **GamePane** with another
+- *SceneManager getInstance()*: returns the instance of the class
+
+
+## Object diagrams								
+Author(s): `Torstein Thomassen`
+
+This object diagram shows an example of objects with different states in our program.
+All the different attributes and variables has to be fulled in with example data.
+
+In this example each team has only three players. This is only for demonstration and will be changed for the final version.
+
+This structure resembles a tree structure with branches and leafs. The tree structure is commonly used in many software design. It's an efficient and a good way to structure the relation between objects. 
+
+This creates a hierarchy where a parent can only directly affect the children. E.g the **Match** can't affect the **Player** 'directly', only the **Team** objects.       
+
+`Object diagram over the Match Class`
+![](imgs/ass2%20obj%20diagram.png)
```

Comment:
Would have been great if you had modeled the object diagram of a league and not just a single match, remember these diagrams should present your feature as a whole not just parts of it